### DISCUSSION TITLE
Repackaged and published certain runtime targets

### DISCUSSION THREAD

**Discussion by crazycrank** [dotnet/runtime#76790] [General]
Hi 

I'm not sure if this is the right place for this, so please feel free to redirect me if necessary.

I've been working with analyzers and source generators the past couple of weeks, and what I struggled the most with was how to correctly pack these, so end users won't be having any issues and warnings whatsoever, and I just couldn't figure it out.

In the end I looked at dotnet/runtime repository, specifically at the System.Text.Json project, and with the help of the GetAnalyzerPackFiles target (defined [here](https://github.com/dotnet/runtime/blob/main/src/libraries/Directory.Build.props)) and the IncludeAnalyzersInPackage combined with the AnalyzerReference tag (defined [here](https://github.com/dotnet/runtime/blob/main/eng/packaging.targets)) I was finally able to get it to work.

Since I've seen dozens of posts and issues struggling with similar issues as me, I decided it would be a good idea if these targets would be available as nuget packages for the public and quickly created two nuget packages for use: 
* https://www.nuget.org/packages/Cranks.Targets.GetAnalyzerPackFiles/
* https://www.nuget.org/packages/Cranks.Targets.AnalyzerReference/
* GitHUb repo: https://github.com/crazycrank/Cranks.Targets

I wanted to make sure that it's okay to release the packages with the source code from this repository. The packages are released under the MIT license and it's mentioned that the sourcecode is from the dotnet/runtime repo.
If this is a problem, I will immediatly remove the projects again and make the repo private. Otherwise I hope to be able to help other members of the community improve their experience in write their own analyzers and source generators.

Best,
Max

**Comment by am11** [comment]
devloop has a package called [`NuGetizer`](https://www.nuget.org/packages/NuGetizer) for this purpose. See https://github.com/agocke/serde-dn/tree/f7ee6e0/src/pack for examples.
